docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+    status = ml_pipeline_start (handle);
+    status = ml_pipeline_src_get_handle (handle, "srcx", &srchandle);
+  ```
+
+  With `srchandle`, we can check the information of input tensors.
+
+  ```bash
+    ml_tensors_info_h info;
+
+    status = ml_pipeline_src_get_tensors_info (srchandle, &info);
+  ```
+
+  According to the `info`, the input tensor data can be filled.
+  ```bash
+    ml_tensors_data_h data;
+    
+    status = ml_tensors_data_create (info, &data);
+
+    for (i = 0; i < 10; i++) {
+      uintarray1[i] = (uint8_t *) g_malloc (4);
+      uintarray1[i][0] = i + 4;
+      uintarray1[i][1] = i + 1;
+      uintarray1[i][2] = i + 3;
+      uintarray1[i][3] = i + 2;
+    }
+
+    status = ml_tensors_data_set_tensor_data (data, 0, uintarray1[0], 4);
+
+    /* setiing the policy of raw data pointer */
+    status = ml_pipeline_src_input_data (srchandle, data, ML_PIPELINE_BUF_POLICY_DO_NOT_FREE);
+  ```
+
+  After configuring it, release the handle is required.
+
+  ```bash
+    status = ml_pipeline_src_release_handle (srchandle);
+  ```
+
+* #### Sink <a name="sinkelement"></a>
+  In case of getting the output tensor raw data, the configuration of data sink element is required.
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc num-buffers=3 ! videoconvert ! tensor_converter ! appsink name=sinkx");
+  ```
+
+  Through `ml_pipeline_sink_register ()`, `appsink` element with the name `sinkx` become reachable.
+  ```bash
+    int status;
+    ml_pipeline_h handle;
+    ml_pipeline_sink_h sinkhandle;
+
+    status = ml_pipeline_sink_register (handle, "sinkx", sink_callback, user_data, &sinkhandle);
+  ```
+
+  `sink_callback ()` is called whenever appsink, `sinkx`, receives data. with [`data`](#tensorsdata), [`info`](#tensorsinfo), we can get the output data easily, and `user_data` can be used in the callback.
+
+  ```bash
+    typedef void (*ml_pipeline_sink_cb) (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data);
+  ```
+
+  After using it, the `sinkhandler` has to be released through `ml_pipeline_sink_unregister ()`
+  ```bash
+    status = ml_pipeline_sink_unregister (sinkhandle);
+  ```
+
+
+* #### Valve <a name="valveelement"></a>
+  When the stream of a pipeline needs to be controlled, this element is useful.
+
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! videoscale ! video/x-raw,format=RGBx,width=16,height=16,framerate=10/1 ! tensor_converter ! queue ! valve name=valve1");
+    int status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  ```
+
+  `valve1` is the name of the valve of the pipeline, and it is opened as a default. This valve can be controlled with getting `ml_pipeline_valve_h`.
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_valve_h valve1;
+
+    status = ml_pipeline_valve_get_handle (handle, "valve1", &valve1);
+  ```
+
+  After start pipeline, the stream of a pipeline can be controlled with valve.
```

Comment:
Please rephrase to:
After you start a pipeline, you can control the stream of the pipeline with a valve: